x/lockup/keeper/lock.go

```diff
@@ -73,7 +76,36 @@ func (k Keeper) deleteLockRefs(ctx sdk.Context, lockRefPrefix []byte, lock types
 	return nil
 }
 
+func (k Keeper) addSyntheticLockRefs(ctx sdk.Context, lockRefPrefix []byte, lock types.PeriodLock) error {
```

Comment:
Not really an actionable comment, but I'm generally a bit worried about the complexity growing of all our separate lock indices. Is there a way we can simplify / re-use a lot of the logic?

Reply:
I was thinking a lot but we need different entries list for synthetic lockups to prevent conflict with native lockups.